lib/tbd/geo.rb

```diff
@@ -357,22 +362,82 @@ def properties(surface = nil, argh = {})
+      # fenestration areas. TBD also relies on OpenStudio subsurface type
+      # assignment, with resulting TBD tags being a bit more concise, e.g.:
+      #
+      #   - :window includes "FixedWindow" and "OperableWindow"
+      #   - :door includes "Door", "OverheadWindow" and "GlassDoor"
+      #     ... a (roof) access roof hatch should be assigned as a "Door"
+      #   - :skylight includes "Skylight", "TubularDaylightDome", etc.
+      #
+      type = :skylight
+      type = :window if typ.include?("window") # operable or not
+      type = :door   if typ.include?("door")   # fenestrated or not
+
+      # In fact, ANY subsurface other than :window or :door is tagged as
+      # :skylight, e.g. a glazed floor opening (CN, Calgary, Tokyo towers). This
+      # happens to reflect OpenStudio default initialization behaviour. For
+      # instance, a subsurface added to an exposed (horizontal) floor in
+      # OpenStudio is automatically assigned a "Skylight" type. This is similar
+      # to the auto-assignment of (opaque) walls, roof/ceilings and floors
+      # (based on surface tilt) in OpenStudio.
+      #
+      # When it comes to major thermal bridging, ASHRAE 90.1 (2022) makes a
+      # clear distinction between "vertical fenestration" (a defined term) and
+      # all other subsurfaces. "Vertical fenestration" would include both
+      # instances of "Window", as well as "GlassDoor". It would exclude however
+      # a non-fenestrated "door" (another defined term), like "Door" &
+      # "OverheadDoor", as well as skylights. TBD tracks relevant subsurface
+      # attributes via a handful of boolean variables:
+      glazed   = type == :door && typ.include?("glass")   # fenestrated door
+      tubular  =                  typ.include?("tubular") # dome or diffuser
+      domed    =                  typ.include?("dome")    # (tubular) dome
+      unhinged = false                                    # (tubular) dome
+
+      # It would be tempting (and simple) to have TBD further validate whether a
+      # "GlassDoor" is actually integrated within a (vertical) wall. The
+      # automated type assignment in OpenStudio is very simple and reliable (as
+      # discussed in the preceding paragraphs), yet users can nonetheless reset
+      # this explicitly. For instance, while a vertical surface may indeed be
+      # auto-assigned "Wall", a modeller can just as easily reset its type as
+      # "Floor". Although OpenStudio supports 90.1 rules by default, it's not
+      # enforced. TBD retains the same approach: for whatever osbcur reason a
+      # modeller may decide (and hopefully the "authority having jurisdiction"
+      # may authorize) to reset a wall as a "Floor" or a roof skylight as a
+      # "GlassDoor", TBD maintains the same OpenStudio policy. Either OpenStudio
+      # (and consequently EnergyPlus) sub/surface type assignment is reliable,
+      # or it is not.
```

Comment:
On the fence with such policies of not enforcing (obvious) checks, e.g. verticality of a "Wall" or a "FixedWindow". This is one of several internal instances where a gem/measure like TBD _could_ deploy more stringent checks than OpenStudio itself. But deploying greater stringency than OpenStudio shifts the responsibility to gem/measure developers - not necessarily a great idea. My 2-cents.